File: dist_servers/Server3.py
import socket
import threading
import time

PORT = 5002  # Server2'nin portu

def handle_client(client_socket):
    try:
        # İstemciden gelen mesajı al
        message = client_socket.recv(1024).decode('utf-8')
        print(f"Server3({threading.get_ident()}) üzerinde istemciden gelen mesaj: {message}")

        # İstemciye cevap gönder
        response = "55 TAMM"
        client_socket.sendall(response.encode('utf-8'))
    except Exception as e:
        print(f"İstemci işleme hatası: {e}")
    finally:
        client_socket.close()

def ping_other_server(host, port):
    while True:
        try:
            with socket.create_connection((host, port), timeout=5):
                print(f"{host} üzerindeki {port} portuna ping gönderildi")
        except Exception:
            print(f"{host} üzerindeki {port} portuna ping başarısız oldu, yeniden deniyor...")
        
        time.sleep(10)  # 10 saniye bekle

def start_server():
    server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    server_socket.bind(('localhost', PORT))
    server_socket.listen()
    print(f"Server2 {PORT} portunda çalışıyor")

    # Diğer sunuculara ping gönder
    threading.Thread(target=ping_other_server, args=("localhost", 5000)).start()  # Server1'e ping
    threading.Thread(target=ping_other_server, args=("localhost", 5001)).start()  # Server3'e ping

    # İstemci bağlantılarını dinle
    try:
        while True:
            client_socket, _ = server_socket.accept()
            threading.Thread(target=handle_client, args=(client_socket,)).start()
    except Exception as e:
        print(f"Sunucu hatası: {e}")
    finally:
        server_socket.close()

if __name__ == "__main__":
    start_server()

File: dist_servers/Server3.java
import java.io.*;
import java.net.*;
import java.time.Instant;

public class Server3 {
    private static final int PORT = 5002; // Server3 portu
    private static final int[] OTHER_PORTS = {5000, 5001}; // Diğer sunucuların portları

    public static void main(String[] args) {
        // Diğer sunucularla bağlantı kur
        for (int port : OTHER_PORTS) {
            new Thread(() -> connectToServer("localhost", port)).start();
        }

        // Sunucuyu başlat
        try (ServerSocket serverSocket = new ServerSocket(PORT)) {
            System.out.println("Server3 " + PORT + " portunda çalışıyor...");
            while (true) {
                Socket clientSocket = serverSocket.accept();
                new Thread(() -> handleClient(clientSocket)).start();
            }
        } catch (IOException e) {
            System.out.println("Server3 hata: " + e.getMessage());
        }
    }

    private static void connectToServer(String host, int port) {
        try (Socket socket = new Socket(host, port)) {
            System.out.println("Server3, " + host + ":" + port + " ile bağlantı kurdu.");
        } catch (IOException e) {
            System.out.println("Server3, " + host + ":" + port + " bağlantı kuramadı.");
        }
    }

    private static void handleClient(Socket clientSocket) {
        try (BufferedReader in = new BufferedReader(new InputStreamReader(clientSocket.getInputStream()));
             PrintWriter out = new PrintWriter(clientSocket.getOutputStream(), true)) {

            String message = in.readLine();
            System.out.println("Server3'den gelen mesaj: " + message);

            if ("STRT".equals(message)) {
                out.println("Demand: STRT, Response: YEP");
            } else if ("CPCTY".equals(message)) {
                long timestamp = Instant.now().getEpochSecond();
                out.println("server3_status: 1000, timestamp: " + timestamp);
            } else {
                out.println("Demand: " + message + ", Response: NOP");
            }

        } catch (IOException e) {
            System.out.println("Server3 istemci hatası: " + e.getMessage());
        }
    }
}
